test/it/TestDescribeIndex.cpp

```diff
@@ -0,0 +1,52 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "TypeUtils.h"
+#include "mocks/MilvusMockedTest.h"
+
+using ::milvus::StatusCode;
+using ::milvus::proto::milvus::DescribeIndexRequest;
+using ::milvus::proto::milvus::DescribeIndexResponse;
+using ::testing::_;
+using ::testing::AllOf;
+using ::testing::ElementsAreArray;
+using ::testing::Property;
+
+TEST_F(MilvusMockedTest, DescribeIndexFoo) {
+    milvus::ConnectParam connect_param{"127.0.0.1", server_.ListenPort()};
+    client_->Connect(connect_param);
+
+    const std::string collection_name = "test_collection";
+    const std::string field_name = "test_field";
+
+    DescribeIndexRequest request;
+    request.set_collection_name(collection_name);
+    request.set_field_name(field_name);
+
+    EXPECT_CALL(service_, DescribeIndex(_,
+                                        AnyOf(Property(&DescribeIndexRequest::collection_name, collection_name),
+                                              Property(&DescribeIndexRequest::field_name, field_name)),
+                                        _))
+        .WillOnce([](::grpc::ServerContext*, const DescribeIndexRequest*, const DescribeIndexResponse*) {
+            return ::grpc::Status{};
```

Comment:
If you want to pass the test like `EXPECT_EQ(index_desc.FieldName(), ...)`, you need to set the expected response here.

Reply:
How Do I set the response? There aren't any available methods to do that.

Reply:
`.WillOnce([](::grpc::ServerContext*, const DescribeIndexRequest*, const DescribeIndexResponse*) {` 
should be 
`.WillOnce([](::grpc::ServerContext*, const DescribeIndexRequest*, DescribeIndexResponse* response) {`

Reply:
Okay got it .

Reply:
But I still can't change the response as index_descriptions provided in the response is const type.

Reply:
the proto files don't have any methods to set the index_descriptions in a response

Reply:
> index_descriptions

For changing proto generated member structure, you should use mutable_member(), like, response->mutable_index_descriptions().

>
> Bar* mutable_foo(): Returns a pointer to the mutable Bar object that stores the field's value. If the field was not set prior to the call, then the returned Bar will have none of its fields set (i.e. it will be identical to a newly-allocated Bar). After calling this, has_foo() will return true and foo() will return a reference to the same instance of Bar.
> 

more ref: https://developers.google.com/protocol-buffers/docs/reference/cpp-generated

Reply:
I got it. It wasn't mutable methods but add methods. Since the response didn't have any it was setting an auto index_descriptor with empty values.